javaslang/src/main/java/javaslang/collection/Comparators.java

```diff
@@ -47,7 +47,11 @@ private Comparators() {
         long serialVersionUID = 1L;
 
         static <T> SerializableComparator<T> of(Comparator<T> comparator) {
-            return comparator::compare;
+            if (comparator instanceof SerializableComparator) {
```

Comment:
This protection is necessary. But I ask myself why it has been called with a SerializableComparator as argument. Once a plain Comparator is changed to be a SerializableComparator it should not be necessary to call SerializableComparator.of again. At least I didn't have to when implementing the RedBlackTree...